### PR TITLE
Allow to configure trusted proxies and headers using config options

### LINKED ISSUE #13890: [FrameworkBundle] allow configuring trusted proxies using semantic conf…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#37357
| PR author(s) | @nicolas-grekas
| Merged in    | 5.2-dev

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #13890.

If you agree with this, I'll add the XML and PHP config formats to the example.

### CODE REVIEW

**Comment by xabbuh** [comment]
looks good to me 👍

**Comment by javiereguiluz** [comment]
I've updated this with XML and PHP config.

@xabbuh in the XSD file of these new options (https://github.com/symfony/symfony/pull/37357/files#diff-e7427ecf018e4434c1645712ebe715f6) they are defined as "strings". Is this correct? How can I use a PHP constant as part of a string in XML config? Thanks!

**Comment by xabbuh** [comment]
I think you are right and we need to remove the constant examples from the XML code block.

File: deployment/proxies.rst [hunk 1]
```diff
@@ -24,30 +24,73 @@ Solution: ``setTrustedProxies()``
...
 The Request object has several ``Request::HEADER_*`` constants that control exactly
 *which* headers from your reverse proxy are trusted. The argument is a bit field,
 so you can also pass your own value (e.g. ``0b00110``).
 
+.. versionadded:: 5.2
+
+    The feature to configure trusted proxies and headers with ``trusted_proxies``
+    and ``trusted_headers`` options was introduced in Symfony 5.2. In earlier
+    Symfony versions you needed to use the ``Request::setTrustedProxies()``
+    method in the ``public/index.php``file.
```
Comment (wouterj) [inline deployment/proxies.rst]:
```suggestion
    method in the ``public/index.php`` file.
```

**Comment by vstelmakh** [comment]
Would be nice to have this somewhen merged. I wasted some time investigating that `framework.trusted_proxies` exist again. 🙂

**Comment by wouterj** [comment]
Thank you Javier! I'm very sorry, I somehow got 3 separate conflicts and ended up with a commit that someone assigned me as author instead of you.

**Comment by javiereguiluz** [comment]
@wouterj no worries 🙏   Thanks for merging!

**Comment by vstelmakh** [comment]
Wow, that was fast. Thanks to all involved 👍

**Pull request merged**